config-model/src/main/java/com/yahoo/vespa/model/builder/xml/dom/DomAdminV4Builder.java

```diff
@@ -100,8 +102,7 @@ private NodesSpecification createNodesSpecificationForLogserver() {
         if (deployState.getProperties().useDedicatedNodeForLogserver() &&
                 context.getApplicationType() == ConfigModelContext.ApplicationType.DEFAULT &&
                 deployState.isHosted() &&
-                deployState.zone().system() == SystemName.cd &&
-                deployState.zone().environment() == Environment.dev)
+                Flags.ENABLE_LOGSERVER.bindTo(deployState.flagSource()).value())
```

Comment:
In order to get the flag value for a particular application, you need to tell the flag what application this is for before value() with a call to `with(Dimension.APPLICATION_ID, anApplicationId.serializedForm())`